main.go

```diff
@@ -68,8 +69,24 @@ func main() {
 	flag.Parse()
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+	restConfig := ctrl.GetConfigOrDie()
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+	kubeClient, err := client.New(restConfig, client.Options{})
```

Comment:
Can we reorganise this code and client from the manager?
```
hasKeda, err := controllers.HasExistingKedaInstallation(mgr.GetClient(), setupLog)
```

Reply:
I did it this way for two reasons:
- Die as soon as possible before starting the manager.
- The manager client is a cached reader. If it's used right after the manager is created, the cache might not be populated quickly enough and the resources are not found.

Reply:
The problem with this line is that right now we have two Kubernetes clients in the main.go file. Marcin's proposition was to use clients from the manager to not duplicate these fat structures. Maybe we can think about another solution? But first I can answer to following arguments:
*  `Die as soon as possible before starting the manager` - not exactly. We can create mgr object first, then use the client from the manager in the `HasExistingKedaInstallation` method, and then start the mgr. In this case, we can still die as soon as possible before starting the manager.
* `The manager client is a cached reader` - so maybe a good option would be to pass the custom client builder to the mgr? I see the options for the mgr.New method we can define a custom ClientBuildFunc where the default one is defined [here](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/cluster/cluster.go#L259). Based on this and on the existing client, we can define cache for the client we created a few lines above. But definitely simpler would we be to use the client from the manager.